packages/p2p/src/components/buy-sell/buy-sell-form.jsx

```diff
@@ -86,6 +101,18 @@ const BuySellForm = props => {
         }
     };
 
+    const onSubmitAdTransaction = async (...args) => {
+        const updated_market_rate = await fetchCurrentMarketRate();
+        if (market_rate_on_mount !== updated_market_rate) {
+            buy_sell_store.setShouldShowPopup(false);
+            setTimeout(() => {
+                buy_sell_store.setShowRateChangedPopup(true);
+            }, 250);
```

Comment:
```suggestion
            setTimeout(() => {
                buy_sell_store.setShowRateChangedPopup(true);
            }, 250);
```
👀  why `setTimeout` ?

Reply:
This was done to solve a UI glitch. the animation takes 250ms. So I am delaying the execution of code to open the next modal by 250ms